code/core/src/core-server/dev-server.ts

```diff
@@ -53,20 +52,13 @@ export async function storybookDevServer(options: Options) {
   app.use(getAccessControlMiddleware(core?.crossOriginIsolated ?? false));
   app.use(getCachingMiddleware());
 
-  getMiddleware(options.configDir)(router);
-
-  app.use(router);
+  getMiddleware(options.configDir)(app);
 
   const { port, host, initialPath } = options;
   invariant(port, 'expected options to have a port');
   const proto = options.https ? 'https' : 'http';
   const { address, networkAddress } = getServerAddresses(port, host, proto, initialPath);
 
-  const listening = new Promise<void>((resolve, reject) => {
-    // @ts-expect-error (Following line doesn't match TypeScript signature at all 🤔)
-    server.listen({ port, host }, (error: Error) => (error ? reject(error) : resolve()));
-  });
```

Comment:
What's this replaced with? or why is this no longer needed?

Reply:
I originally misread the Polka API, but I've now brought this back almost identical - further down.

https://github.com/storybookjs/storybook/pull/29230/files#diff-a93733d92246593f521ce12d8457e6af4629f8ae5312e184fabffc99a94ee703R119-R122